src/test/datascience/notebook/executionHelper.ts

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {
+    CancellationToken,
+    NotebookCell,
+    NotebookCellExecution,
+    NotebookCellExecutionSummary,
+    NotebookCellKind,
+    NotebookCellOutput,
+    NotebookCellOutputItem,
+    NotebookDocument,
+    NotebookRange,
+    TextDocument,
+    Uri,
+    workspace
+} from 'vscode';
+import { IKernelController } from '../../../kernels/types';
+import { InteractiveWindowView, JupyterNotebookView, PYTHON_LANGUAGE } from '../../../platform/common/constants';
+import { ReadWrite } from '../../../platform/common/types';
+import { MockNotebookDocuments } from './helper';
+
+export function createKernelController(controllerId = '1'): IKernelController {
+    return {
+        // eslint-disable-next-line @typescript-eslint/no-use-before-define
+        createNotebookCellExecution: (cell: NotebookCell) => new TestNotebookCellExecution(cell),
+        id: controllerId
+    };
+}
+class TestNotebookCellExecution implements NotebookCellExecution {
```

Comment:
Used for faster tests, we just have mocks for some of the classes
With this change we can now bring down test times from 6 minutes to <30s

Basically we're bypassing VS Code API to edit & update notebook documents and the like.
We know they work, hence no point going through all of that API.

We have some basic tests that run through VS Code end to end, but for a majority of the tests the plan is to just use mocks.